lib/membrane/rtp/packet.ex

```diff
@@ -73,28 +78,36 @@ defmodule Membrane.RTP.Packet do
         payload_type: payload_type,
         timestamp: timestamp,
         csrcs: for(<<csrc::32 <- csrcs>>, do: csrc),
-        extension: extension
+        extension: extension,
+        has_padding?: has_padding == 1,
+        total_header_size: 32 + 32 + 32 + 32 * csrcs_cnt + header_extension_size
       }
 
-      {:ok, %__MODULE__{header: header, payload: payload}}
+      {:ok,
+       %__MODULE__{
+         header: header,
+         payload: if(parse_payload?, do: payload, else: unmodified_payload)
+       }}
     else
       :error -> {:error, :malformed_packet}
     end
   end
 
-  def parse(_binary), do: {:error, :malformed_packet}
+  def parse(_binary, _parse_payload?), do: {:error, :malformed_packet}
 
-  defp parse_header_extension(binary, header_present?)
-  defp parse_header_extension(binary, false), do: {:ok, {nil, binary}}
+  defp parse_header_extension(binary, header_present?, parse_payload?)
+  defp parse_header_extension(binary, false, _parse_payload?), do: {:ok, {nil, 0, binary}}
 
   defp parse_header_extension(
-         <<profile_specific::16, data_len::16, data::binary-size(data_len)-unit(32),
-           rest::binary>>,
-         true
+         <<profile_specific::binary-size(2), data_len::16, data::binary-size(data_len)-unit(32),
+           rest::binary>> = unmodified_payload,
+         true,
+         parse_payload?
        ) do
     extension = %Header.Extension{profile_specific: profile_specific, data: data}
-    {:ok, {extension, rest}}
+
+    {:ok, {extension, 32 + data_len * 32, if(parse_payload?, do: rest, else: unmodified_payload)}}
```

Comment:
Before reading a document about how binaries work internally I thought that it could somehow be expensive copying the binary but since it creates a view over the original binary I guess we can pass `rest` in every case. Header size will be easier to calculate.

Reply:
ok, let's go for it